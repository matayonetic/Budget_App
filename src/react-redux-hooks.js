import {
  useSelector as originalUseSelector,
  useDispatch as originalUseDispatch,
} from "react-redux";

export const useDispatch = () => originalUseDispatch();
export const useSelector = (state) => originalUseSelector(state);
