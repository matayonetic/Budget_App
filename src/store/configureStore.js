import { legacy_createStore as createStore, combineReducers } from "redux";

// Reducers
import expensesReducer from "../reducers/expensesReducer";
import filtersReducer from "../reducers/filtersReducer";

// Store
const configureStore = () => {
  const store = createStore(
    combineReducers({
      expenses: expensesReducer,
      filters: filtersReducer,
    }),
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  );
  return store;
};

export { configureStore as default };
