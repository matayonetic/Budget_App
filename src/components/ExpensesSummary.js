import React from "react";
import { useSelector } from "../react-redux-hooks";
import numeral from "numeral";
import getExpenseCount from "../selectors/getExpensesCount";
import getExpensesTotal from "../selectors/getExpensesTotal";
import getVisibleExpenses from "../selectors/getVisibleExpenses";

const ExpensesSummary = () => {
  const expenses = useSelector((state) =>
    getVisibleExpenses(state.expenses, state.filters)
  );
  const eCount = getExpenseCount(expenses);
  const eTotal = getExpensesTotal(expenses);

  return (
    <>      
      {eCount <= 1 && (
        <h2>
          Viewing {eCount} expense with a total of{" "}
          {numeral(eTotal / 100).format("$0,0.00")}
        </h2>
      )}
      {eCount >= 2 ? (
        <h2>
          Viewing {eCount} expense(s) with a total of{" "}
          {numeral(eTotal / 100).format("$0,0.00")}
        </h2>
      ) : (
        <></>
      )}
    </>
  );
};

export default ExpensesSummary;
