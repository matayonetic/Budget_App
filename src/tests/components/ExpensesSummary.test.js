import React from "react";
import ExpensesSummary from "../../components/ExpensesSummary";
import { shallow } from "enzyme";

import expenses from "../fixtures/expenses";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import * as ReactReduxHooks from "../../react-redux-hooks";

// For One Expense
test("Render ExpenseSummary correctly for one expense", () => {
  const store = configureStore([thunk])([expenses[0]]);
  jest
    .spyOn(ReactReduxHooks, "useSelector")
    .mockImplementation(() => store.getState());
  const wrapper = shallow(<ExpensesSummary />);
  expect(wrapper).toMatchSnapshot();
});

// For multiple Expenses
test("Render ExpenseSummary correctly for multiple expenses", () => {
    const store = configureStore([thunk])(expenses);
    jest
      .spyOn(ReactReduxHooks, "useSelector")
      .mockImplementation(() => store.getState());
    const wrapper = shallow(<ExpensesSummary />);
    expect(wrapper).toMatchSnapshot();
  });
