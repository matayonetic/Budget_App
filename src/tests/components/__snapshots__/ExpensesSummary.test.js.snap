// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ExpenseSummary correctly for multiple expenses 1`] = `
<Fragment>
  <h2>
    Viewing 
    3
     expenses with a total of
     
    $0.06
  </h2>
</Fragment>
`;

exports[`Render ExpenseSummary correctly for one expense 1`] = `
<Fragment>
  <h2>
    Viewing 
    1
     expense with a total of
     
    $0.01
  </h2>
</Fragment>
`;
